### PR TITLE
[SYCL] Make Level-Zero interop API SYCL-2020 compliant for queue, event, and kernel_bundle(was program).

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Review by alexbatashev — APPROVED** [review]
These tests should be eventually updated as well, I think: https://github.com/intel/llvm-test-suite/tree/intel/SYCL/Basic/interop

File: sycl/include/CL/sycl/backend/level_zero.hpp [hunk 1]
```diff
@@ -152,6 +176,7 @@ T make(const std::vector<device> &DeviceList,
 // Construction of SYCL program.
 template <typename T, typename detail::enable_if_t<
                           std::is_same<T, program>::value> * = nullptr>
+__SYCL_DEPRECATED("Use SYCL-2020 sycl::make_kernel_bundle free function")
```
[the remark this answers is not in the document]
Comment (bader) [reply sycl/include/CL/sycl/backend/level_zero.hpp:179]: @smaslov-intel, please, fix all instances.

**Comment by bader** [comment]
> These tests should be eventually updated as well, I think: https://github.com/intel/llvm-test-suite/tree/intel/SYCL/Basic/interop

@smaslov-intel, fix `sycl/test/basic_tests/interop-level-zero-2020.cpp` as well.
Please, do not modify file permissions for sycl/include/CL/sycl/backend/level_zero.hpp.

File: sycl/include/CL/sycl/backend/level_zero.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/include/CL/sycl/backend/level_zero.hpp]: fixed

**Comment by bader** [comment]
@smaslov-intel, please, sync with the tip of the branch. There are merge conflicts.

File: sycl/include/CL/sycl/backend/level_zero.hpp [hunk 2]
```diff
@@ -13,4 +13,246 @@
 __SYCL_WARNING("CL/sycl/backend/level_zero.hpp usage is deprecated, include "
                "sycl/ext/oneapi/backend/level_zero.hpp instead")
 
+<<<<<<< HEAD
```
Comment (bader) [inline sycl/include/CL/sycl/backend/level_zero.hpp:16]: Please, fix merge conflicts.

Comment (smaslov-intel) [reply sycl/include/CL/sycl/backend/level_zero.hpp]: fixed

**Comment by smaslov-intel** [comment]
> Failed Tests (4):
[2021-09-08T21:04:17.580Z]   SYCL :: OnlineCompiler/online_compiler_L0.cpp
[2021-09-08T21:04:17.580Z]   SYCL :: Plugin/interop-level-zero-keep-ownership.cpp
[2021-09-08T21:04:17.580Z]   SYCL :: Plugin/interop-level-zero.cpp
[2021-09-08T21:04:17.580Z]   SYCL :: Regression/cache_test.cpp

The tests in Precommit testing failed due to these prior change: https://github.com/intel/llvm/pull/4432/files#diff-bad45c4e6d46c89d807c929b6adbffdb70993c102647c58b41fb0534e2195727

The tests need to be adjusted. @dm-vodopyanov : are you doing that?
It will also break existing customer's codes, is that acknowledged?

**Comment by smaslov-intel** [comment]
> The tests need to be adjusted. @dm-vodopyanov : are you doing that?

@dm-vodopyanov : ping
@bader : are you aware of this?

File: sycl/include/sycl/ext/oneapi/backend/level_zero.hpp [hunk 3]
```diff
@@ -197,9 +224,41 @@ context make_context<backend::level_zero>(
       BackendObject.Ownership == ext::oneapi::level_zero::ownership::keep);
 }
 
-namespace __SYCL2020_DEPRECATED("use 'ext::oneapi::level_zero' instead")
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/ext/oneapi/backend/level_zero.hpp:200]: Please don't remove this

**Review by alexbatashev — APPROVED** [review]

**Pull request merged**